app/controllers/concerns/authenticates_user.rb

```diff
@@ -0,0 +1,14 @@
+# Enforces user authentication for a controller and stores the authenticated user in `Current`
+module AuthenticatesUser
```

Comment:
To anticipate a conversation we've had before: the default naming convention for concerns tends to be `_____able`, but I do like to diverge from that in two cases:
- something sets up a relationship (where I'd go for `Has_____`), or
- non-model concerns, which tend to be a more diverse bunch (like here)

`DoesThing` (`AuthenticatesUser`) feels natural here to me but I'm open for debate!

I'll point to [Gitlab's controller concerns](https://gitlab.com/gitlab-org/gitlab/-/tree/master/app/controllers/concerns) as some prior art, with concerns like:
- `AuthenticatesWithTwoFactor`
- `ChecksCollaboration`
- `RendersNotes`

Reply:
To be fair `-able` doesn't really fit with Authenticate, so I have no concerns (see what I did there?) about the name `AuthenticatesUser`.

Thanks for explaining your choice though 🎉